.github/workflows/scripts/generate_yml_files.py

```diff
@@ -0,0 +1,316 @@
+  if b2s_file_found:
+    print("     - DirectB2S file found!")
+  else:
+    print("     - DirectB2S file NOT found")
+  yml_text = "vps_b2s_ids: \n"
+  for b2s_file_id in b2s_file_ids:
+    yml_text += "     - " + b2s_file_id + "\n"
+  return yml_text
+
+def find_rom_file_info(urls, table_data):
+  rom_file_found = False
+  rom_file_info = []
+  for rom_file in table_data.get("romFiles", []):
+    for url in urls:
+      for file_url in rom_file["urls"]:
+        if clean_url(url) == clean_url(file_url["url"]):
+          rom_file_found = True
+          rom_file_info.append({
+            "id": rom_file["id"],
+            "version": rom_file["version"],
+            "authors": ", ".join(rom_file.get("authors", []))
+          })
+          break  # Stop searching once a match is found
+  if rom_file_found:
+    print(f"     - Rom file found!")
+  else:
+    print(f"     - Rom file NOT found")
+  yml_text = "vps_rom_ids: \n"
+  for rom_file in rom_file_info:
+    yml_text += "     - version: " + rom_file["version"] + "\n"
+    yml_text += "       authors: " + rom_file["authors"] + "\n"
+  return yml_text
+
+def parse_download_links(readme):
+  url_pattern = r"\(([^()]+)\)"
+  urls = re.findall(url_pattern, readme)
+  yml_text = ""
+  # get the block of JSON for the table my matching a URL
+  for url in urls:
+    try:
+      parsed_url = urllib.parse.urlparse(url)
+      if parsed_url.scheme and parsed_url.netloc:
+        if "profile/" not in url and "&showuser" not in url:
+          url = clean_url(url)
+    except ValueError:
+      pass  # Ignore invalid URLs
+
+    table_data = find_table_in_VPSDB(url)
+    if table_data:
+      yml_text = find_vpx_file_ids(urls, table_data)
+      yml_text += find_directb2s_file_ids(urls, table_data)
+      yml_text += find_rom_file_info(urls, table_data)
+      break    
+  else:
+    print(f"Table ID not found!!")
+  return yml_text
+
+# process README.md file and generate yml file
+def process_READMEmd_file(entry, dir):
+   # open a table-specific JSON file (eventually iterate)
+  folder_path = os.path.dirname(entry.path)
+  with open(folder_path + '/' "README.md", "r", errors="replace") as f:
+  #readme = open(folder_path + '/' "README.md", "r", errors="replace")
+    readme = f.read()
+    print('     - README.md file data loaded')
+ 
+    yml_text = write_yml_header()
+ 
+    # parse table name from first line of .md
+    yml_text += parse_and_set_table_name(readme)
+
+    # get tester from git history
+    yml_text += get_earliest_commit_owner(project_directory, folder_path)
+
+    # get FPS
+    yml_text += get_FPS(readme)
+    yml_text += get_special_instructions(readme)
+    yml_text += parse_download_links(readme)
+
+    #write output .yml files to temp folder for now = we can put them in folders later using this...
+      #yml = open(folder_path + '/' + entry.name +  '.yml', "w")
+
+    yml_file_path = project_directory + "/.github/workflows/temp_yml/" + dir +  '.yml'
```

Comment:
this has to be outside of the workflows directory due to permissions... change it to /.wizard/yaml